lib/application.js

```diff
@@ -0,0 +1,180 @@
+	 */
+	this.create = function (params, callback) {
+		return client.makeRequest({
+			path   : "applications",
+			method : "POST",
+			body   : params
+		})
+		.then(function (response) {
+			var application = params;
+			var location = response.headers.location;
+			var applicationId = location.substring(location.lastIndexOf("/") + 1);
+			application.id = applicationId;
+			return application;
+		})
+		.asCallback(callback);
+	};
+
+	/**
+	 * Get an application.
+	 * @param  {String} applicationId The ID of the application to get.
+	 * @param  {Function} [callback] A callback for the application.
+	 * @return {ApplicationResponse} A promise for the application.
+	 */
+	this.get = function (applicationId, callback) {
+		return client.makeRequest({
+			path   : "applications/" + applicationId,
+			method : "GET"
+		})
+		.then(function (response) {
+			return response.body;
+		})
+		.asCallback(callback);
+	};
+
+	/**
+	 * Make changes to an application.
+	 * @param  {String} applicationId The ID of the application to modify.
+	 * @param {Object} params Parameters for creating a new call
+	 * @param {String} [params.name] A name you choose for this application.
+ 	 * @param {String} [params.incomingCallUrl] A URL where call events will be sent for an inbound call.
+ 	 * This is the endpoint where the Application Platform will send all call events.
+	 * Either incomingCallUrl or incomingMessageUrl is required.
+ 	 * @param {String} [params.incomingCallUrlCallbackTimeout] Determine how long should the platform wait
+ 	 * for incomingCallUrl's response before timing out in milliseconds.
+ 	 * @param {String} [params.incomingCallFallbackUrl] The URL used to send the callback
+ 	 * event if the request to incomingCallUrl fails.
+ 	 * @param {String} [params.incomingMessageUrl] A URL where message events will be sent for an inbound message.
+ 	 * This is the endpoint where the Application Platform will send all message events.
+ 	 * Either incomingMessageUrl or incomingCallUrl is required.
+	 * @param {Number} [params.incomingMessageUrlCallbackTimeout] Determine how long should the platform wait for
+ 	 * incomingMessageUrl's response before timing out in milliseconds.
+	 * @param {String} [params.incomingMessageFallbackUrl] The URL used to send the callback event if
+	 * the request to incomingMessageUrl fails.
+	 * @param {String} [params.callbackHttpMethod] Determine if the callback event should be sent via HTTP GET
+ 	 * or HTTP POST. Values are "get" or "post", default: "post".
+ 	 * @param {Boolean} [params.autoAnswer] Determines whether or not an incoming call should be
+ 	 * automatically answered. Default value is 'true'.
+ 	 * @param {Function} [callback] A callback for the list of applications
+	 */
+	this.update = function (applicationId, params, callback) {
+		return client.makeRequest({
+			path   : "applications/" + applicationId,
+			method : "POST",
+			body   : params
+		})
+		.then(function () {
```

Comment:
an empty .then() does the same as not calling .then(). You can remove these 3 lines